### ISSUE TITLE
too much undefined reference

### ISSUE THREAD

**Issue by changtong1996** [Visa-Research/volepsi#4]
when I run 
python3 build.py -DCOPROTO_ENABLE_BOOST=ON -DCOPROTO_ENABLE_OPENSSL=ON
there are all undefined reference like this 

/usr/local/include/libOTe/TwoChooseOne/Iknp/IknpOtExtSender.h:43: undefined reference to `vtable for osuCrypto::IknpOtExtSender'
/usr/local/include/libOTe/TwoChooseOne/Iknp/IknpOtExtSender.h:43: undefined reference to `vtable for osuCrypto::IknpOtExtSender'
../volePSI/libvolePSI.a(RsCpsi.cpp.o): In function `volePSI::RsCpsiSender::send(nonstd::span_lite::span<osuCrypto::block, 18446744073709551615ul>, osuCrypto::MatrixView<unsigned char>, volePSI::RsCpsiSender::Sharing&, coproto::Socket&)::{lambda(macoro::FrameBase<macoro::detail::task_promise<void, true> >*)#1}::operator()(macoro::FrameBase<macoro::detail::task_promise<void, true> >*)':
/home/test/Desktop/volepsi/volePSI/RsCpsi.cpp:130: undefined reference to `osuCrypto::BetaCircuit::~BetaCircuit()'
/home/test/Desktop/volepsi/volePSI/RsCpsi.cpp:130: undefined reference to `osuCrypto::BetaCircuit::~BetaCircuit()'

**Comment by ladnir** [comment]
Have you previously installed libOTe?

**Comment by changtong1996** [comment]
Yes, I suspect the two libraries will conflict. But when I changed to  a new environment, it will fail too.
`/volepsi/out/build/linux/frontend$ ./frontend -u
 0 - Paxos_buildRow_Test           Passed   0ms
 1 - Paxos_solve_Test              Passed   22ms
 2 - Paxos_solve_u8_Test           Passed   7ms
 3 - Paxos_solve_mtx_Test          Passed   11ms
 4 - Paxos_invE_Test               Passed   0ms
 5 - Paxos_invE_g3_Test            Passed   0ms
 6 - Paxos_solve_gap_Test          Passed   2ms
 7 - Paxos_solve_rand_Test         Passed   0ms
 8 - Paxos_solve_rand_gap_Test     Passed   3ms
 9 - Baxos_solve_Test              Passed   0ms
10 - Baxos_solve_mtx_Test          Passed   0ms
11 - Baxos_solve_par_Test          Passed   17ms
12 - Baxos_solve_rand_Test         Passed   29ms
13 - SilentTripleGen_test          Passed   32ms
14 - generateBase_test           Segmentation fault (core dumped)
`

**Comment by ladnir** [comment]
OK, yeah I'm not sure about this segfaults... Someone else also got it. Only impacts circuit psi.

You can try to debug it and see where the segfaults is. Or enable as an by defining VOLE_PSI_ENABLE_ASAN=true, i.e.

`python3 build.py -DVOLE_PSI_ENABLE_ASAN=true`

**Comment by changtong1996** [comment]
Thanks for your reply! I'll try this.

**Comment by changtong1996** [comment]
It seems that the running time is too long on my computer(ubuntu 18).
The stage of generating OPRF occupied much time.
Is there some settings to accelerated?

> 
> ./frontend  -u psi
> 19 - Psi_Rs_empty_test             Passed   61ms   --set size is default 13243
> 20 - Psi_Rs_partial_test           Passed   55ms   --set size is default 13243
> 21 - Psi_Rs_full_test              Passed   2310ms   --set size is 2^20  
> 22 - Psi_Rs_multiThrd_test         Passed   1988ms     --set size is 2^20
> 23 - Psi_Rs_mal_test               Passed   57ms
> 25 - filebase_psi_bin_Test         Passed   71ms
> 26 - filebase_psi_csv_Test         Passed   126ms
> 27 - filebase_psi_csvh_Test        Passed   133ms
>

**Comment by ladnir** [comment]
Run the benchmark code if you want running times. Make sure to enable silver
